cloud-operator/templates/role.yaml

```diff
@@ -21,4 +21,15 @@ rules:
 - apiGroups: [""]
   resources: ["secrets"]
   resourceNames: ["hubble-relay-client-certs"] # Name of the Secret
-  verbs: ["get"]
\ No newline at end of file
+  verbs: ["get"]
+---
+# Role to read secrets in the GKE managed observability namespace
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
Have you validated that this Helm chart now works in non-GKE environments?
I suspect that trying to create a resource in namespace `gke-managed-dpv2-observability` will fail if that namespace doesn't exist, which is the case in any non-GKE cluster.
The creation of this resource must be conditional on the existence of the `gke-managed-dpv2-observability` namespace.
Wrap this with:
```
{{- if (lookup "v1" "Namespace" "gke-managed-dpv2-observability" "") }}
...
{{- end }}
```